src/hooks/useScreenSize.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import { useTheme } from '../components/Theme';
+
+export const useScreenSize = () => {
+  const [ mobile, setMobile ] = useState(false);
+
+  const { theme } = useTheme();
+  const handleResize = () => {
+    if (window.innerWidth > theme.mobileView) {
+      setMobile(false);
+    } else {
+      setMobile(true);
+    }
+  };
+
+  useEffect( () => {
+    window.addEventListener('resize', handleResize)
+    return () => {
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [handleResize]);
```

Comment:
Since `handleResize` is variable, which is recreated every render, `resize` listener will be removed/added every render. Is this something required? Because I think you can use empty deps `[]` and it will work in the same way, but listener will be attached once

Reply:
updated!